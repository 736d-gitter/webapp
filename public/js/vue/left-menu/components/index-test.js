const mount = require('../../__test__/vuex-mount');
const { default: Index } = require('../../../../../public/js/vue/left-menu/components/index.vue');

describe('left-menu index', () => {
  describe('all state', () => {
    it('matches snapshot', () => {
      const { wrapper } = mount(Index, {}, store => {
        store.state.leftMenuState = 'all';
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('people state', () => {
    it('matches snapshot', () => {
      const { wrapper } = mount(Index, {}, store => {
        store.state.leftMenuState = 'people';
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('search state', () => {
    it('matches snapshot', () => {
      const { wrapper } = mount(Index, {}, store => {
        store.state.leftMenuState = 'search';
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('pinning and expanding', () => {
    it('not pinned and collapse', () => {
      const { wrapper } = mount(Index, {}, store => {
        store.state.leftMenuPinnedState = false;
        store.state.leftMenuExpandedState = false;
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('not pinned and expanded', () => {
      const { wrapper } = mount(Index, {}, store => {
        store.state.leftMenuPinnedState = false;
        store.state.leftMenuExpandedState = true;
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('calls store action "toggleLeftMenu" after mouse leaves', () => {
      const { wrapper, stubbedActions } = mount(Index, {}, store => {
        store.state.leftMenuPinnedState = false;
        store.state.leftMenuExpandedState = true;
      });
      wrapper.find({ ref: 'root' }).trigger('mouseleave');

      expect(stubbedActions.toggleLeftMenu).toHaveBeenCalledWith(
        expect.anything(),
        false,
        undefined
      );
    });
  });

  describe('mobile', () => {
    it('mobile matches snapshot', () => {
      const { wrapper } = mount(Index, {}, store => {
        store.state.isMobile = true;
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('nli (not logged in)', () => {
    it('matches snapshot', () => {
      const { wrapper } = mount(Index, {}, store => {
        store.state.isLoggedIn = false;
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('mobile matches snapshot', () => {
      const { wrapper } = mount(Index, {}, store => {
        store.state.isMobile = true;
        store.state.isLoggedIn = false;
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('dark-theme', () => {
    it('matches snapshot', () => {
      const { wrapper } = mount(Index, {}, store => {
        store.state.darkTheme = true;
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
