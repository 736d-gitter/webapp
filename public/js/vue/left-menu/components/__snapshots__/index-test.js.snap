// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`left-menu index all state matches snapshot 1`] = `
<div
  class="root js-left-menu-root logged-in"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="body"
  >
    <section
      class="layout-minibar"
    >
      <menu-bar-body-stub />
    </section>
     
    <section
      class="body-main-menu layout-main-menu"
    >
      <div
        class="hide"
      >
        
        initialTestValue
      
      </div>
       
      <h2
        class="room-list-title"
      >
        All conversations
      </h2>
       
      <!---->
       
      <room-list-stub
        rooms=""
      />
    </section>
  </section>
</div>
`;

exports[`left-menu index dark-theme matches snapshot 1`] = `
<div
  class="root js-left-menu-root logged-in dark-theme"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="body"
  >
    <section
      class="layout-minibar"
    >
      <menu-bar-body-stub />
    </section>
     
    <section
      class="body-main-menu layout-main-menu"
    >
      <div
        class="hide"
      >
        
        initialTestValue
      
      </div>
       
      <h2
        class="room-list-title"
      >
        All conversations
      </h2>
       
      <!---->
       
      <room-list-stub
        rooms=""
      />
    </section>
  </section>
</div>
`;

exports[`left-menu index mobile mobile matches snapshot 1`] = `
<div
  class="root js-left-menu-root mobile logged-in"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="body"
  >
    <section
      class="layout-minibar"
    >
      <menu-bar-body-stub />
    </section>
     
    <section
      class="body-main-menu layout-main-menu"
    >
      <div
        class="hide"
      >
        
        initialTestValue
      
      </div>
       
      <h2
        class="room-list-title"
      >
        All conversations
      </h2>
       
      <!---->
       
      <room-list-stub
        rooms=""
      />
    </section>
  </section>
</div>
`;

exports[`left-menu index nli (not logged in) matches snapshot 1`] = `
<div
  class="root js-left-menu-root"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="nli-body"
  >
    <h2
      class="nli-primary-heading"
    >
      Where communities thrive
    </h2>
     
    <br />
     
    <ul
      class="nli-info-block"
    >
      <li
        class="nli-info-block-item"
      >
        Join over 
        <strong>
          1.5M+ people
        </strong>
      </li>
       
      <li
        class="nli-info-block-item"
      >
        Join over 
        <strong>
          100K+ communities
        </strong>
      </li>
       
      <li
        class="nli-info-block-item"
      >
        Free 
        <strong>
          without limits
        </strong>
      </li>
       
      <li
        class="nli-info-block-item"
      >
        Create 
        <strong>
          your own community
        </strong>
      </li>
    </ul>
     
    <ul
      class="nli-info-block"
    >
      <li
        class="nli-info-block-item"
      >
        <a
          href="/explore"
        >
          Explore more communities
        </a>
      </li>
    </ul>
  </section>
</div>
`;

exports[`left-menu index nli (not logged in) mobile matches snapshot 1`] = `
<div
  class="root js-left-menu-root mobile"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="nli-body"
  >
    <h2
      class="nli-primary-heading"
    >
      Where communities thrive
    </h2>
     
    <br />
     
    <ul
      class="nli-info-block"
    >
      <li
        class="nli-info-block-item"
      >
        Join over 
        <strong>
          1.5M+ people
        </strong>
      </li>
       
      <li
        class="nli-info-block-item"
      >
        Join over 
        <strong>
          100K+ communities
        </strong>
      </li>
       
      <li
        class="nli-info-block-item"
      >
        Free 
        <strong>
          without limits
        </strong>
      </li>
       
      <li
        class="nli-info-block-item"
      >
        Create 
        <strong>
          your own community
        </strong>
      </li>
    </ul>
     
    <ul
      class="nli-info-block"
    >
      <li
        class="nli-info-block-item"
      >
        <a
          href="/explore"
        >
          Explore more communities
        </a>
      </li>
    </ul>
  </section>
</div>
`;

exports[`left-menu index people state matches snapshot 1`] = `
<div
  class="root js-left-menu-root logged-in"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="body"
  >
    <section
      class="layout-minibar"
    >
      <menu-bar-body-stub />
    </section>
     
    <section
      class="body-main-menu layout-main-menu"
    >
      <div
        class="hide"
      >
        
        initialTestValue
      
      </div>
       
      <!---->
       
      <h2
        class="room-list-title"
      >
        People
      </h2>
       
      <room-list-stub
        rooms=""
      />
    </section>
  </section>
</div>
`;

exports[`left-menu index pinning and expanding not pinned and collapse 1`] = `
<div
  class="root js-left-menu-root logged-in unpinned"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="body"
  >
    <section
      class="layout-minibar"
    >
      <menu-bar-body-stub />
    </section>
     
    <section
      class="body-main-menu layout-main-menu"
    >
      <div
        class="hide"
      >
        
        initialTestValue
      
      </div>
       
      <h2
        class="room-list-title"
      >
        All conversations
      </h2>
       
      <!---->
       
      <room-list-stub
        rooms=""
      />
    </section>
  </section>
</div>
`;

exports[`left-menu index pinning and expanding not pinned and expanded 1`] = `
<div
  class="root js-left-menu-root logged-in unpinned expanded"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="body"
  >
    <section
      class="layout-minibar"
    >
      <menu-bar-body-stub />
    </section>
     
    <section
      class="body-main-menu layout-main-menu"
    >
      <div
        class="hide"
      >
        
        initialTestValue
      
      </div>
       
      <h2
        class="room-list-title"
      >
        All conversations
      </h2>
       
      <!---->
       
      <room-list-stub
        rooms=""
      />
    </section>
  </section>
</div>
`;

exports[`left-menu index search state matches snapshot 1`] = `
<div
  class="root js-left-menu-root logged-in"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="body"
  >
    <section
      class="layout-minibar"
    >
      <menu-bar-body-stub />
    </section>
     
    <section
      class="body-main-menu layout-main-menu"
    >
      <div
        class="hide"
      >
        
        initialTestValue
      
      </div>
       
      <search-body-stub />
    </section>
  </section>
</div>
`;
